src/main/java/org/javarosa/form/api/FormEntryModel.java

```diff
@@ -466,10 +465,8 @@ private void incrementHelper(Vector<Integer> indexes, Vector<Integer> multiplici
                 multiplicities.setElementAt(0, i);
                 elements.setElementAt(parent.getChild(curIndex + 1), i);
 
-                if (repeatStructure == REPEAT_STRUCTURE_NON_LINEAR) {
-                    if (elements.lastElement() instanceof GroupDef && ((GroupDef)elements.lastElement()).getRepeat()) {
-                        multiplicities.setElementAt(TreeReference.INDEX_REPEAT_JUNCTURE, multiplicities.size() - 1);
-                    }
+                if (elements.lastElement() instanceof GroupDef && ((GroupDef)elements.lastElement()).getRepeat()) {
```

Comment:
@wspride Can you explain why this is a needed change at the formentry model level? CommCare Android does this currently, so i'm surprised to see that a change is needed in core, rather than at the platform/presentation layer.